MIGRATION.md

```diff
@@ -0,0 +1,239 @@
+# Migrating to the `cloudant-java-sdk` library
+This document is to assist in migrating from the `java-cloudant` (coordinates: `com.cloudant:cloudant-client`) to the newly supported [`cloudant-java-sdk`](https://github.com/IBM/cloudant-java-sdk) (coordinates: `com.ibm.cloud:cloudant`).
+
+## Initializing the client connection
+There are several ways to create a client connection in `cloudant-java-sdk`:
+1. [Environment variables](https://github.com/IBM/cloudant-java-sdk#authentication-with-environment-variables)
+2. [External configuration file](https://github.com/IBM/cloudant-java-sdk#authentication-with-external-configuration)
+3. [Programmatically](https://github.com/IBM/cloudant-java-sdk#programmatic-authentication)
+
+[See the README](https://github.com/IBM/cloudant-java-sdk#code-examples) for code examples on using environment variables.
+
+## Other differences
+1. Fetching the Database object first before performing additional operations is not required. For example, in the case of updating a document you would first call `getDocument` to fetch and then `putDocument` to update.
+1. Model classes are used instead of POJOs. See examples in the [POJO usage in the `cloudant-java-sdk` library](#pojo-usage-in-the-new-library).
+1. Sending and receiving byte responses is available for operations that accept user-defined documents or return user-defined documents, document projections or map/reduce data. See [the Raw IO section](https://github.com/IBM/cloudant-java-sdk#raw-io) of `cloudant-java-sdk` README or the [Bypass the document model and use the `asStream` methods section](#3-bypass-the-document-model-and-use-the-asstream-methods) for more details.
+1. There is no pagination support for views. Examples coming soon.
+
+### POJO usage in the new library
+
+Since the new library supports models, this guide will provide you three ways to migrated your POJOs from `java-cloudant`.
+
+Let's start with a simple POJO:
+
+```java
+public class POJO {
+    private String _id;
+    private String name;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public String toString() {
+        return _id + ": " + name;
+    }
+}
+```
+The code block below uses the POJO in a simple read and write implementation using
+Cloudant Query to read the object and then updating the document in the database.
+
+```java
+//... set up the service client and the database
```

Comment:
Remove prefix dots
```suggestion
// set up the service client and the database
```
I think it may be good to just add the code for creating a client and database.

Reply:
I think anyone who wants to migrate knows how to do all this. Anyway, it's all in README and javadoc.